runtime/src/runtime/fvm.rs

```diff
@@ -292,56 +292,20 @@ where
         &self.blockstore
     }
 
-    fn send(
+    fn send_generalized(
         &self,
         to: &Address,
         method: MethodNum,
         params: Option<IpldBlock>,
         value: TokenAmount,
-    ) -> Result<Option<IpldBlock>, ActorError> {
+        gas_limit: Option<u64>,
+        flags: SendFlags,
+    ) -> Result<Response, ErrorNumber> {
         if self.in_transaction {
-            return Err(actor_error!(assertion_failed; "send is not allowed during transaction"));
-        }
-        match fvm::send::send(to, method, params, value, None, SendFlags::empty()) {
-            Ok(ret) => {
-                if ret.exit_code.is_success() {
-                    Ok(ret.return_data)
-                } else {
-                    Err(ActorError::checked(
-                        ret.exit_code,
-                        format!(
-                            "send to {} method {} aborted with code {}",
-                            to, method, ret.exit_code
-                        ),
-                        ret.return_data,
-                    ))
-                }
-            }
-            Err(err) => Err(match err {
-                // Some of these errors are from operations in the Runtime or SDK layer
-                // before or after the underlying VM send syscall.
-                ErrorNumber::NotFound => {
-                    // This means that the receiving actor doesn't exist.
-                    // TODO: we can't reasonably determine the correct "exit code" here.
-                    actor_error!(unspecified; "receiver not found")
-                }
-                ErrorNumber::InsufficientFunds => {
-                    // This means that the send failed because we have insufficient funds. We will
-                    // get a _syscall error_, not an exit code, because the target actor will not
-                    // run (and therefore will not exit).
-                    actor_error!(insufficient_funds; "not enough funds")
-                }
-                ErrorNumber::LimitExceeded => {
-                    // This means we've exceeded the recursion limit.
-                    // TODO: Define a better exit code.
-                    actor_error!(assertion_failed; "recursion limit exceeded")
-                }
-                err => {
-                    // We don't expect any other syscall exit codes.
-                    actor_error!(assertion_failed; "unexpected error: {}", err)
-                }
-            }),
+            return Err(ErrorNumber::IllegalOperation);
```

Comment:
Hmm, error numbers come from the VM, but this is just a library-level check. But it wouldn't be worth doing a `Result<Result<...>>` so on balance I don't see a better way. Maybe just drop a comment noting how we're slightly bending things here.

Reply:
Done.